### ISSUE TITLE
Unable to join 1.12.2 server on a 1.13 client through BungeeCord with ViaVersion.

### ISSUE THREAD

**Issue by mibby** [MylesIsCool/ViaVersion#981]
**Bungee Environment**
Waterfall dev 218
https://destroystokyo.com/ci/view/all/job/Waterfall/
ViaVersion dev 224
https://ci.viaversion.com/view/All/job/ViaVersion/

**Server Environment**
Paper dev 1547 _(Spigot 1.12.2)_
https://destroystokyo.com/ci/view/all/job/Paper/
ProtocolSupport dev 242 _(Last Spigot 1.12.2 supported build)_
https://build.true-games.org/job/ProtocolSupport

When I try connecting through BungeeCord to a 1.12.2 Spigot server with a 1.13 client, I get disconnected from the server with Spigot's ```outdated-server``` message.

I have also tried ViaVersion-DEV dev 154 on BungeeCord but unfortunately run into the same issue.

Is ViaVersion not compatible with ProtocolSupport? I know the main Spigot page says to use ViaVersion on BungeeCord as a workaround, but it doesn't seem to let me connect with a higher client version doing so. ViaBackwards is missing some features like remapping maps to older versions so I use ProtocolSupport for server client support <1.12.

**Comment by MylesIsCool** [comment]
You have to manually set the protocol versions to work around this via the ViaVersion config (and disable auto-updating).

It's not compatible when they're on Spigot togehter in 1.12, in the future please provide dumps as they can be vital to finding the issue.

**Comment by mibby** [comment]
How would I manually set the protocol versions? I already have auto-update disabled.

```
# Should ViaVersion check for updates?
checkforupdates: false
```

```
bungee-servers:
  default: 47
  Server: -1
```

**Comment by mibby** [comment]
@MylesIsCool Do you mean forcing the server protocol higher than what the server version actually is?

i.e
```
# 
# ----------------------------------------------------------#
#                     BUNGEE OPTIONS                       #
# ----------------------------------------------------------#
# 
# BungeeCord allows you to have different server versions inside.
# Instead of you entering all the versions of these servers, we can ping them.
# 
# What interval would you like us to ping at? (in seconds)
# Use -1 to disable.
bungee-ping-interval: 60
# If the above is enabled, should we save the info to the config (in the section below)
bungee-ping-save: false
# To get a servers protocol, ViaVersion will do the following:
# Look for the server in the following section, then look for the last ping if bungee-ping is enabled
# otherwise use default.
# 
# The format for the following is:
#  servername: protocolversion
# You can find protocol ids on http://wiki.vg/Protocol_version_numbers
# It will fallback to the default option if none found.
bungee-servers:
  default: 47
  Server: 399
```

**Comment by MylesIsCool** [comment]
@mibby No ensuring it's the real version as ViaVersion has issues detecting the protocol version of ProtocolSupport servers

**Comment by mibby** [comment]
So just match Server: # with the protocol version the server is running on? i.e. ```Server: 340``` and then ViaVersion will allow versions higher than 340 to connect through it?

**Comment by MylesIsCool** [comment]
@mibby Yep

**Comment by mibby** [comment]
Alright, thanks. :) I'll give it a test. However with it set to -1 and BungeeCord probing the server, I am able to have 1.8 and 1.12 clients connect fine - just 1.13 clients having issues. Hopefully manually setting the server protocol will fix it!

**Comment by MylesIsCool** [comment]
@mibby 

You're going to have to provide /viaversion dump for further support.

**Comment by mibby** [comment]
I linked it above. :(

**Comment by mibby** [comment]
@MylesIsCool Any ideas? This is the output of viaversion with debug enabled while trying to connect.
https://paste.ubuntu.com/p/RS3F8MvX9H/

**Comment by MylesIsCool** [comment]
Apologies didn't realize you had edited the comments. The issue is down to BungeeCord rewriting the plugin messages from protocolsupport, because it uses a pipe symbol this isn't compatible with 1.13.

Ideally, BungeeCord should rewrite the pipe symbol as this issue could also happen if a BungeeCord plugin used it for messaging. I'm not sure if we have any plans to fix this ourself (as the logic is quite deep in Bungee). 

Possible solutions:

- Use ViaRewind / ViaBackwards instead of ProtocolSupport (this will allow down to 1.7 and is fine)
- Update to 1.13 and use ProtocolSupport without ViaVersion (they changed the plugin channel in future versions so it'd work fine)
- Modify ProtocolSupport to remove the | from the code eg. https://github.com/ProtocolSupport/ProtocolSupport/blob/6f3b8b878e967d64b03f1a81685ffe2bba54f4a8/src/protocolsupport/protocol/packet/handler/AbstractLoginListenerPlay.java#L123 (obviously, you'd have to look for all instances of payloads like this and also compile the 1.12 version)
- Modify BungeeCord to remove the | from the code

**Comment by MylesIsCool** [comment]
Just to clarify finally, you are running the latest build (I'd assume so), https://ci.viaversion.com/job/ViaVersion/

**Comment by mibby** [comment]
* Unfortunately ViaRewind/ViaBackwards has a few remapping issues, which is why I'm stuck with using ProtocolSupport instead. i.e. https://github.com/Gerrygames/ViaRewind/issues/82
* While I would update to 1.13, there are a handful of plugins I'm still waiting on that are still not compatible and some cosmetic & gameplay bugs within the 1.13 client that are not worth putting up with. I was going to wait for 1.13.1's release to fix some serious issues and client performance before updating fully.
* Is it just the character ```|``` that is causing the issue? I can't find any instance of BungeeCord having it, unless you mean BungeeCord should be modified to strip all unsupported characters in the plugin messaging channel?

I'll look into cloning ProtocolSupport's repo and seeing if I can revert commits for 1.12 version compiling without that symbol. However if it is also better handled by BungeeCord and should be rewritten, I'll create a new ticket for Bungee. 

I am running the latest ViaVersion build, yes. Just swapping between the latest ViaVersion and ViaVersion-DEV since -DEV branch shouldn't have much difference, being it is 1.13.1 > 1.13 remapping.

**Comment by TheCompieter** [comment]
@mibby can you provide me with a download because i have the same isue Thanks
